crates/persistent-storage/src/lib.rs

```diff
@@ -0,0 +1,41 @@
+use base64::{engine::general_purpose, Engine as _};
+use ethers::prelude::*;
+use ethers_signers::WalletError;
+use reqwest::{multipart, Client};
+use serde::Deserialize;
+
+#[derive(Deserialize)]
+#[allow(non_snake_case)]
+struct IpfsCreateResponse {
+    Hash: String,
+}
+
+#[derive(Debug, thiserror::Error)]
+pub enum IPFSError {
+    #[error("Authentication message signature error: {0:?}")]
+    AuthSigning(#[from] WalletError),
+    #[error("Invalid Peer ID: {0:?}")]
+    Network(#[from] reqwest::Error),
+}
+
+pub async fn write_to_ipfs(client: Client, file: String) -> Result<String, IPFSError> {
+    let auth_key = get_auth_key().await?;
+    let form = multipart::Form::new().text("file", file);
+    Ok(client
+        .post("https://crustipfs.xyz/api/v0/add")
```

Comment:
Probably makes sense to make this configurable. Could be another field of the struct.